File: app/src/main/java/com/saliim/absensimobile/loginRegister/LoginActivity.java
package com.saliim.absensimobile.loginRegister;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;
import androidx.appcompat.widget.Toolbar;

import com.saliim.absensimobile.Admin.AdminActivity;
import com.saliim.absensimobile.MainActivity;
import com.saliim.absensimobile.R;
import com.saliim.absensimobile.api.API;
import com.saliim.absensimobile.model.loginUser.LoginUser;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class LoginActivity extends AppCompatActivity {

    private Toolbar toolbar;
    public static String name;
    public static String id;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        toolbar = findViewById(R.id.toolbars);
        setSupportActionBar(toolbar);
        getSupportActionBar().setTitle("Login User");

        final EditText etLoginUsername = findViewById(R.id.et_login_username);
        final EditText etLoginPassword = findViewById(R.id.et_login_password);
        Button loginSubmit = findViewById(R.id.btn_login_submit);
        TextView pindahReg = findViewById(R.id.pindah_register);

        loginSubmit.setOnClickListener(v -> {
            String inLogUn = etLoginUsername.getText().toString();
            String inLogPw = etLoginPassword.getText().toString();
            API.loginUsers(inLogUn, inLogPw).enqueue(new Callback<LoginUser>() {
                @Override
                public void onResponse(Call<LoginUser> call, Response<LoginUser> response) {
                    Log.d("login", response.body().toString());

                    String status = response.body().getResult();
                    if (status.equals("1")){
                        name = response.body().getUser().getNama();
                        id = response.body().getUser().getId();
                        String level = response.body().getUser().getVslevel();

                        if (level.equals("normal user")){
                            startActivity(new Intent(LoginActivity.this, MainActivity.class));

                            Toast.makeText(LoginActivity.this, "berhasil login" , Toast.LENGTH_SHORT).show();
                            finish();
                        }else if (level.equals("admin")){
                            startActivity(new Intent(LoginActivity.this, AdminActivity.class));

//                                Toast.makeText(LoginActivity.this, "selamat datang admin" , Toast.LENGTH_SHORT).show();
                            finish();
                        }

                    }else{
                        Toast.makeText(LoginActivity.this, "periksa kembali name dan passwordmu" , Toast.LENGTH_SHORT).show();
                    }
                }

                @Override
                public void onFailure(Call<LoginUser> call, Throwable t) {
                    Toast.makeText(LoginActivity.this, "gagal" , Toast.LENGTH_SHORT).show();
                }
            });
        });

        pindahReg.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(LoginActivity.this, RegisterActivity.class));
            }
        });

    }
}

File: app/src/main/java/com/saliim/absensimobile/loginRegister/LoginActivity.kt
package com.saliim.absensimobile.loginRegister

import android.content.Context
import androidx.appcompat.app.AppCompatActivity

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.widget.Toolbar
import com.google.gson.Gson

import com.saliim.absensimobile.admin.AdminActivity
import com.saliim.absensimobile.MainActivity
import com.saliim.absensimobile.R
import com.saliim.absensimobile.api.API
import com.saliim.absensimobile.model.loginUser.LoginUser
import com.saliim.absensimobile.subAdmin.SubAdminActivity

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class LoginActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        val etLoginUsername = findViewById<EditText>(R.id.et_login_username)
        val etLoginPassword = findViewById<EditText>(R.id.et_login_password)
        val loginSubmit = findViewById<Button>(R.id.btn_login_submit)
        val pindahReg = findViewById<TextView>(R.id.pindah_register)

        loginSubmit.setOnClickListener { v ->
            val inLogUn = etLoginUsername.text.toString()
            val inLogPw = etLoginPassword.text.toString()
            API.loginUsers(inLogUn, inLogPw).enqueue(object : Callback<LoginUser> {
                override fun onResponse(call: Call<LoginUser>, response: Response<LoginUser>) {
                    Log.d("login", response.body()!!.toString())

                    val user = response.body()

                    getSharedPreferences(MY_LOGIN_PREF, Context.MODE_PRIVATE).edit().putString(MY_LOGIN_PREF_KEY, Gson().toJson(user)).apply()

                    if (user!!.result == "1") {

                        val level = response.body()!!.user.vslevel

                        val savedUser = Gson().fromJson<LoginUser>(this@LoginActivity.getSharedPreferences(MY_LOGIN_PREF, Context.MODE_PRIVATE)
                                .getString(MY_LOGIN_PREF_KEY, ""), LoginUser::class.java)

                        id = savedUser.user.id
                        name = savedUser.user.nama

                        when (level) {
                            "normal user" -> {
                                startActivity(Intent(this@LoginActivity, MainActivity::class.java))
                                finish()
                            }
                            "admin" -> {
                                startActivity(Intent(this@LoginActivity, AdminActivity::class.java))
                                finish()
                            }
                            "subAdmin" -> {
                                startActivity(Intent(this@LoginActivity, SubAdminActivity::class.java))
                                finish()
                            }
                        }

                    } else {
                        Toast.makeText(this@LoginActivity, "periksa kembali name dan passwordmu", Toast.LENGTH_SHORT).show()
                    }
                }

                override fun onFailure(call: Call<LoginUser>, t: Throwable) {
                    Toast.makeText(this@LoginActivity, "gagal", Toast.LENGTH_SHORT).show()
                }
            })
        }

        pindahReg.setOnClickListener { startActivity(Intent(this@LoginActivity, RegisterActivity::class.java)) }

    }

    companion object {
        lateinit var name: String
        lateinit var id: String

        const val MY_LOGIN_PREF = "myLoginPref"
        const val MY_LOGIN_PREF_KEY = "loginPrefKey"
    }
}
